### ISSUE TITLE
Question: Why ColorSpectrum has to be squared?

### ISSUE THREAD

**Issue by FrayxRulez** [microsoft/microsoft-ui-xaml#1812]
Hello,
I'm trying to restyle the ColorPicker as I'd like to get something really compact, similar to this:
![image](https://user-images.githubusercontent.com/8152849/71826189-a84f7600-309d-11ea-80fe-1790eee2011a.png)
The concern is that ColorPicker sets the Height of the ColorSpectrum inside to its Width.
My question is: is this really necessary? Would be possible to avoid this?
I would ever go for my own implementation using ColorSpectrum and ColorPickerSlider, but it doesn't seem possible to use ColorPickerSlider as a standalone control.

**Comment by StephenLPeters** [comment]
I think the reason we do this in code behind is because the math we use to layout the color choices is not robust enough to layout correctly with a rectangular footprint.  I think the only way we'd be able to get a rectangle to work is to show less (or more) shades for each hue. I think this would probably be fine, but it not a trivial change. If this is blocking could you open an issue? I doubt it would meet our bar though.

**Comment by llongley** [comment]
It's been a long time since I've looked at it, but if I'm recalling correctly, the limiting factor is that without explicitly setting the height of the ColorSpectrum, it will measure its height as zero, since it doesn't have any internal layout that tells it what height it should otherwise be.  So we explicitly set its height to ensure that doesn't happen, and we set it to make it a square as that's what design called for.

That said, though, I can't at least immediately think of a reason why we couldn't honor an explicitly set height if one is given.  I agree with @StephenLPeters that the best option would be to file an issue requesting the feature of non-square ColorSpectrums, which will put it on our radar as something to consider looking at.
